api/capi/src/nnstreamer-capi-pipeline.c

```diff
@@ -1556,11 +1556,197 @@ ml_pipeline_valve_set_open (ml_pipeline_valve_h h, bool open)
   handle_exit (h);
 }
 
+/********************************************************
+ ** NNStreamer Element Property Control in Pipeline    **
+ ********************************************************/
+/**
+ * @brief Gets an element handle in NNStreamer pipelines to control its properties.
+ */
+int
+ml_pipeline_get_element (ml_pipeline_h pipe, const char *element_name,
+    ml_pipeline_element_h *elm_h)
+{
+  int ret = ML_ERROR_NONE;
+  ml_pipeline_element *elem;
+  ml_pipeline_common_elem *common_elem;
+  ml_pipeline *p = pipe;
+
+  /* Check input parameter */
+  if (pipe == NULL) {
+    ml_loge ("The first argument, pipeline handle, is not valid.");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  if (element_name == NULL) {
+    ml_loge ("The second argument, element name, is not valid.");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  if (elm_h == NULL) {
+    ml_loge ("The argument element handle is not valid.");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+  *elm_h = NULL;
+
+  g_mutex_lock (&p->lock);
+
+  /* 1. Search element in lookup table first */
+  elem = g_hash_table_lookup (p->namednodes, element_name);
+  if (elem == NULL) {
+    /* 2. Search element in pipeline itself */
+    GstElement *gst_elem = gst_bin_get_by_name (GST_BIN (p->element), element_name);
+    if (gst_elem == NULL) {
+      ml_loge ("The element named [%s] is not found in the pipeline", element_name);
+      ret = ML_ERROR_INVALID_PARAMETER;
+      goto unlock_return;
+    }
+
+    /* Caching for next search */
+    elem = construct_element (gst_elem, pipe, element_name, ML_PIPELINE_ELEMENT_COMMON);
+    if (elem == NULL) {
+      ml_loge ("Failed to allocate the internal memory");
+      ret = ML_ERROR_OUT_OF_MEMORY;
+      goto unlock_return;
+    }
+    g_hash_table_insert (p->namednodes, g_strdup (element_name), elem);
+  }
+
+  /* Type checking */
+  if (elem->type != ML_PIPELINE_ELEMENT_COMMON) {
+    ml_loge ("There is an element named [%s] in the pipeline, but it is not a common element", element_name);
+    ret = ML_ERROR_INVALID_PARAMETER;
+    goto unlock_return;
+  }
+
+  common_elem = *elm_h = g_new0(ml_pipeline_common_elem, 1);
+  if (common_elem == NULL) {
```

Comment:
I think it's not necessary

Reply:
Thanks for your comment.

Yes, you are right. Null checking for `g_new0()` is useless. However, to follow a conventional coding style, I'd like to keep this code. Somedays, when updating the memory allocation code, I will remove it.